### PR TITLE
dev,genbzl: add support for generating syntax diagrams

### LINKED ISSUE #84443: Move SQL grammar diagram generation from Makefile to Bazel
In order to successfully remove the Makefile, we have to move the SQL grammar diagram generation process from `make` to `dev`. 

Similar example for reference: https://github.com/cockroachdb/cockroach/pull/83727

### PR DESCRIPTION
**Pull request by ajwerner**
Fixes #84443.

Release note: None

### CODE REVIEW

File: dev [hunk 1]
```diff
@@ -3,7 +3,7 @@
 set -euo pipefail
 
 # Bump this counter to force rebuilding `dev` on all machines.
-DEV_VERSION=41
+DEV_VERSION=42
```
Comment (rickystewart) [inline dev:6]: 42 is the current version, you probably want to rebase and bump this again.

**Review by rickystewart — APPROVED** [review]

File: dev (hunk 1 above)
Comment (ajwerner) [reply dev]: Done

**Comment by ajwerner** [comment]
> @ajwerner I tested this locally and this accomplishes part of it and is already super helpful, but we also need to generate the raw BNF files so the docgen CI job can populate the generated-diagrams repo. I can try that once I get started in Dev Inf. Otherwise, great work and LGTM!

It was trivial to add support for generating the `bnf` files. I've updated the change to also include a `./dev generate bnf` subcommand.

**Review by nickvigilante — APPROVED** [review]
@ajwerner You're awesome.

**Comment by ajwerner** [comment]
TFTR!

bors r+

**Pull request merged**